sdk/data/azcosmos/cosmos_global_endpoint_manager_test.go

```diff
@@ -0,0 +1,302 @@
+	assert.NoError(t, err)
+
+	accountProps, err := gem.GetAccountProperties(context.Background())
+	assert.NoError(t, err)
+
+	expectedAccountProps := accountProperties{
+		ReadRegions:                  nil,
+		WriteRegions:                 nil,
+		EnableMultipleWriteLocations: false,
+	}
+	assert.Equal(t, expectedAccountProps, accountProps)
+}
+
+func TestGlobalEndpointManagerCanUseMultipleWriteLocations(t *testing.T) {
+	srv, close := mock.NewTLSServer()
+	defer close()
+	srv.SetResponse(mock.WithStatusCode(http.StatusOK))
+
+	pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{}, &policy.ClientOptions{Transport: srv})
+
+	client := &Client{endpoint: srv.URL(), pipeline: pl}
+
+	preferredRegions := []string{"West US", "Central US"}
+
+	serverEndpoint, err := url.Parse(srv.URL())
+	assert.NoError(t, err)
+
+	mockLc := newLocationCache(preferredRegions, *serverEndpoint)
+	mockLc.enableMultipleWriteLocations = true
+	mockLc.useMultipleWriteLocations = true
+
+	mockGem := globalEndpointManager{
+		client:              client,
+		preferredLocations:  preferredRegions,
+		locationCache:       mockLc,
+		refreshTimeInterval: 5 * time.Minute,
+	}
+
+	gem, err := newGlobalEndpointManager(client, preferredRegions, 5*time.Minute)
+	assert.NoError(t, err)
+
+	// Multiple locations should be false for default GEM
+	canUseMultipleWriteLocs := gem.CanUseMultipleWriteLocations()
+	assert.False(t, canUseMultipleWriteLocs)
+
+	// Mock GEM with multiple write locations available should show true
+	canUseMultipleWriteLocs = mockGem.CanUseMultipleWriteLocations()
+	assert.True(t, canUseMultipleWriteLocs)
+}
+
+func TestGlobalEndpointManagerConcurrentUpdate(t *testing.T) {
+	countPolicy := &countPolicy{}
+	srv, closeFunc := mock.NewTLSServer()
+	defer closeFunc()
+	srv.SetResponse(mock.WithStatusCode(http.StatusOK))
+
+	westRegion := accountRegion{
+		Name:     "West US",
+		Endpoint: srv.URL(),
+	}
+
+	properties := accountProperties{
+		ReadRegions:                  []accountRegion{westRegion},
+		WriteRegions:                 []accountRegion{westRegion},
+		EnableMultipleWriteLocations: false,
+	}
+
+	jsonString, err := json.Marshal(properties)
+	if err != nil {
+		t.Fatal(err)
+	}
+	srv.SetResponse(mock.WithBody(jsonString))
+
+	pl := azruntime.NewPipeline("azcosmostest", "v1.0.0", azruntime.PipelineOptions{PerCall: []policy.Policy{countPolicy}}, &policy.ClientOptions{Transport: srv})
+	req, err := azruntime.NewRequest(context.Background(), http.MethodGet, srv.URL())
+	assert.NoError(t, err)
+
+	req.SetOperationValue(pipelineRequestOptions{
+		isWriteOperation: true,
+	})
+
+	client := &Client{endpoint: srv.URL(), pipeline: pl}
+
+	gem, err := newGlobalEndpointManager(client, []string{}, 5*time.Second)
+	assert.NoError(t, err)
+
+	// Call update concurrently and see how many times the policy gets called
+	concurrency := 5
+	wg := &sync.WaitGroup{}
+	wg.Add(concurrency)
+
+	finishCh := make(chan struct{})
```

Comment:
`finishCh` is not used for the concurrency, right?

Reply:
yeah it was leftover from some testing I was doing, will clean it up